features/nanostack/mbed-mesh-api/source/WisunInterface.cpp

```diff
@@ -57,6 +60,63 @@ nsapi_error_t WisunInterface::do_initialize()
         }
         _interface->attach(_connection_status_cb);
     }
+
+    // Apply mbed configuration to Wi-SUN
+    configure();
+
+    return NSAPI_ERROR_OK;
+}
+
+nsapi_error_t WisunInterface::configure()
+{
+    int status;
+
+    if (_configured) {
+        // Already configured
+        return NSAPI_ERROR_OK;
+    }
+
+    _configured = true;
+#ifdef MBED_CONF_MBED_MESH_API_WISUN_NETWORK_NAME
+    char network_name[] = {MBED_CONF_MBED_MESH_API_WISUN_NETWORK_NAME};
+    status = set_network_name((char *) &network_name);
+    if (status < 0) {
+        tr_error("Failed to set network name!");
+        return NSAPI_ERROR_PARAMETER;
+    }
+#endif
+
+#if (MBED_CONF_MBED_MESH_API_WISUN_REGULATORY_DOMAIN != 255) || (MBED_CONF_MBED_MESH_API_WISUN_OPERATING_CLASS != 255) || (MBED_CONF_MBED_MESH_API_WISUN_OPERATING_MODE != 255)
+    status = set_network_regulatory_domain(MBED_CONF_MBED_MESH_API_WISUN_REGULATORY_DOMAIN,
+                                           MBED_CONF_MBED_MESH_API_WISUN_OPERATING_CLASS,
+                                           MBED_CONF_MBED_MESH_API_WISUN_OPERATING_MODE);
```

Comment:
This does not look like you are able to call changes on these parameters before calling connect

It was previously collected in structure at creation and could be modified by user calling functions and then before interface up these were called

Reply:
This configuration is made on network interface init. So after that and before connect() is called, it is possible to change the settings similarly as previously. 

There is no need for settings cache on application side, since on the new configuration system, the settings can be programmed directly to Wi-SUN stack after the init.